libs/ledger-live-common/src/hw/actions/installLanguage.ts

```diff
@@ -128,13 +136,22 @@ export const createAction = (
       const sub = impl
         .pipe(
           tap((e: any) => log("actions-install-language-event", e.type, e)),
+          takeUntil(abortScanSignaler),
           scan(reducer, getInitialState()),
         )
-        .subscribe(setState);
+        .subscribe({
+          next: state => {
+            setState(state);
+
+            if (state.poll) return;
+
+            abortScanSignaler.next();
```

Comment:
You'll see if https://github.com/LedgerHQ/ledger-live/pull/5350 fixes your issue - but otherwise that should work